stdlib/public/core/CharacterProperties.swift

```diff
@@ -48,11 +56,65 @@ extension Character {
   ///     // lf.asciiValue == 10
   ///     let crlf = "\r\n" as Character
   ///     // crlf.asciiValue == 10
-  @inlinable
+  @_alwaysEmitIntoClient @_silgen_name("inlineASCIIValue")
   public var asciiValue: UInt8? {
-    if _slowPath(self == "\r\n") { return 0x000A /* LINE FEED (LF) */ }
-    if _slowPath(!_isSingleScalar || _firstScalar.value >= 0x80) { return nil }
-    return UInt8(_firstScalar.value)
+    if !_str._guts.isSmall { return outlineASCIIValue }
+    // This character is backed by a small string. We can check if it is a
+    // single ASCII character (and return its value) just by looking at the
+    // first word. This works because Character is made up of a single
+    // grapheme cluster, the NUL character does not combine, and in a
+    // SmallString all unused bytes must be zero, so we don't need to look
+    // at the count.
+    let firstWord = _str._guts.asSmall._storage.0.littleEndian
+    return firstWord < 128 ? UInt8(firstWord) : firstWord == 0x0a0d ? 0xa : nil
+  }
+  
+  // Mangle this as "isASCII" for ABI compatability with older libraries where
+  // the generic isASCII was inlinable public (so calls will exist, even though
+  // it was usually inlined).
+  @usableFromInline @_silgen_name("$sSJ7isASCIISbvg") @inline(never)
+  internal var outlineIsASCII: Bool {
```

Comment:
For consistency, please follow [the leading underscore rule](https://github.com/apple/swift/blob/main/docs/StandardLibraryProgrammersManual.md#the-leading-underscore-rule) when naming non-public entities.

```suggestion
  internal var _outlinedIsASCII: Bool {
```

(I know this name isn't leaking into the ABI; nevertheless, this is making my hobgoblin twitch. 😅)